src/DynamoCoreWpf/Views/Core/NodeView.xaml

```diff
@@ -47,7 +47,10 @@
 
         <!--  CONTEXT MENU  -->
         <Grid.ContextMenu>
-            <ContextMenu Name="MainContextMenu" x:FieldModifier="public">
+            <ContextMenu Name="MainContextMenu"
+                         x:FieldModifier="public"
+                         Closed="MainContextMenu_OnClosed"
+                         Style="{StaticResource NodeContextMenuStyle}">
```

Comment:
I have two questions here:

1. I think including this style for menuItems in contextMenu.Resources can be optimized - can it also be moved to the sharedResource dictionary and only applied in the nodeContextMenuBuilder and applied there?

2. Why does this style include references to the dismissed badge, is that in the context menu at certain times?

Reply:
1. As far as I can tell this is a dumb WPF thing: 

We can style the `ContextMenu` from a ResourceDictionary fine, but styling the items is a little harder. `ContextMenus` can contain both `MenuItems` and `Separators` (we have both of these in the mockup). 
Telling the `ContextMenu` that its ItemContainerStyle should be a style specifically targeting `MenuItems` won't work and this results in an exception. The `Separators` don't like being given a style that's not for them.

Therefore, I put both the style for Separators and MenuItems within a local resources section, so they get picked up as is appropriate and with no exceptions.

2. Yes it is! It's there at all times.

Reply:
Am investigating whether we can use an ItemContainerStyleSelector to get past #2.

Reply:
I've had no luck with implementing the ItemContainerStyleSelector. It seems like a sensible approach, but I've not seen it being used inside of a ResourceDictionary.